dev/src/index.ts

```diff
@@ -647,31 +648,31 @@ export class Firestore {
    * @returns A QueryDocumentSnapshot for existing documents, otherwise a
    * DocumentSnapshot.
    */
-  snapshot_(
+  snapshot_<T = DocumentData>(
     documentName: string,
     readTime?: google.protobuf.ITimestamp,
     encoding?: 'protobufJS'
-  ): DocumentSnapshot;
-  snapshot_(
+  ): DocumentSnapshot<T>;
+  snapshot_<T = DocumentData>(
     documentName: string,
     readTime: string,
     encoding: 'json'
-  ): DocumentSnapshot;
-  snapshot_(
+  ): DocumentSnapshot<T>;
+  snapshot_<T = DocumentData>(
     document: api.IDocument,
     readTime: google.protobuf.ITimestamp,
     encoding?: 'protobufJS'
-  ): QueryDocumentSnapshot;
-  snapshot_(
+  ): QueryDocumentSnapshot<T>;
+  snapshot_<T = DocumentData>(
     document: {[k: string]: unknown},
     readTime: string,
     encoding: 'json'
-  ): QueryDocumentSnapshot;
-  snapshot_(
+  ): QueryDocumentSnapshot<T>;
+  snapshot_<T = DocumentData>(
     documentOrName: api.IDocument | {[k: string]: unknown} | string,
     readTime?: google.protobuf.ITimestamp | string,
     encoding?: 'json' | 'protobufJS'
-  ): DocumentSnapshot {
+  ): DocumentSnapshot<T> {
```

Comment:
These typings are a little off if I am not mistaken. Don't we always return `DocumentSnapshot<DocumentData>`?